diff --git bindings/pyroot/src/Utility.cxx bindings/pyroot/src/Utility.cxx
index f610b3c..efb95ae 100644
--- bindings/pyroot/src/Utility.cxx
+++ bindings/pyroot/src/Utility.cxx
@@ -555,6 +555,7 @@ int PyROOT::Utility::GetBuffer( PyObject* pyobject, char tc, int size, void*& bu
       (*(bufprocs->bf_getbuffer))( pyobject, &bufinfo, PyBUF_WRITABLE );
       buf = (char*)bufinfo.buf;
       Py_ssize_t buflen = bufinfo.len;
+      PyBuffer_Release( &bufinfo );
 #endif
 
       if ( buf && check == kTRUE ) {
