diff --git bindings/pyroot/src/TPyBufferFactory.cxx bindings/pyroot/src/TPyBufferFactory.cxx
index f98be1e..ab30e8d 100644
--- bindings/pyroot/src/TPyBufferFactory.cxx
+++ bindings/pyroot/src/TPyBufferFactory.cxx
@@ -60,6 +60,7 @@ namespace {
       Py_buffer bufinfo;
       (*(Py_TYPE(self)->tp_as_buffer->bf_getbuffer))( self, &bufinfo, PyBUF_SIMPLE );
       Py_ssize_t nlen = bufinfo.len;
+      PyBuffer_Release( &bufinfo );
 #endif
       if ( nlen != INT_MAX )  // INT_MAX is the default, i.e. unknown actual length
          return nlen;
@@ -100,6 +101,7 @@ namespace {
       Py_buffer bufinfo;
       (*(PyBuffer_Type.tp_as_buffer->bf_getbuffer))( self, &bufinfo, PyBUF_SIMPLE );
       buf = (char*)bufinfo.buf;
+      PyBuffer_Release( &bufinfo );
 #endif
 
       if ( ! buf )
